### PR TITLE
Fix Blob URL origin when scheme is file

### PR DESCRIPTION
**Pull request by izgzhen**
r? @Manishearth

Part of #11131

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12195)

### CODE REVIEW

**Comment by izgzhen** [comment]
The interesting thing about this is that, WebKit will return things like `blob:file:///c68fb2b6-ac47-4e55-80b2-650edaa39b77` while Chrome will return `blob:null/c68fb2b6-ac47-4e55-80b2-650edaa39b77`.

This is not well-specified on spec IIRC and I prefer the WebKit way.

**Comment by Manishearth** [comment]
Please mention the discrepancy between browsers in a comment, otherwise r=me 😄

File: components/script/dom/url.rs [hunk 1]
```diff
@@ -196,6 +196,19 @@ impl URL {
 
         result
     }
+
+    // XXX: change String to FileOrigin
+    /* NOTE(izgzhen): WebKit will return things like blob:file:///XXX
+       while Chrome will return blob:null/XXX
+       This is not well-specified, and I perfer the WebKit way here
+    */
```
Comment (izgzhen) [inline components/script/dom/url.rs]: @Manishearth noted here

Comment (Manishearth) [inline components/script/dom/url.rs]: spelling!

@bors-servo delegate+

Comment (izgzhen) [inline components/script/dom/url.rs]: Oops....thanks

**Pull request merged**